rust/src/ssh/ssh.rs

```diff
@@ -156,20 +160,32 @@ impl SSHState {
             match parser::ssh_parse_record(input) {
                 Ok((rem, head)) => {
                     SCLogDebug!("SSH valid record {}", head);
-                    input = rem;
-                    if head.msg_code == SSH_MSG_NEWKEYS {
-                        hdr.flags = SSHConnectionState::SshStateFinished;
-                        if ohdr.flags >= SSHConnectionState::SshStateFinished {
-                            unsafe {
-                                AppLayerParserStateSetFlag(
-                                    pstate,
-                                    APP_LAYER_PARSER_NO_INSPECTION
+                    match head.msg_code {
+                        parser::MessageCode::SshMsgKexinit if unsafe {SSHHasshIsEnabled()} => {
```

Comment:
I'll check, but `rs_ssh_register_parser` is not a good place, because if hassh is used in rules, but not enabled in `suricata.yaml`, then it will not work.